libsolidity/formal/BMC.cpp

```diff
@@ -691,19 +719,31 @@ void BMC::inlineFunctionCall(FunctionCall const& _funCall)
 	}
 	else
 	{
-		initializeFunctionCallParameters(*funDef, symbolicArguments(_funCall, m_currentContract));
+		initializeFunctionCallParameters(*_funDef, symbolicArguments(_funDef, _calledExpr, _funType, _arguments));
 
 		// The reason why we need to pushCallStack here instead of visit(FunctionDefinition)
-		// is that there we don't have `_funCall`.
-		pushCallStack({funDef, &_funCall});
+		// is that there we don't have `_callStackExpr`.
+		pushCallStack({_funDef, &_callStackExpr});
 		pushPathCondition(currentPathConditions());
 		auto oldChecked = std::exchange(m_checked, true);
-		funDef->accept(*this);
+		_funDef->accept(*this);
 		m_checked = oldChecked;
 		popPathCondition();
 	}
 
-	createReturnedExpressions(_funCall, m_currentContract);
+	createReturnedExpressions(_funDef, _callStackExpr);
+}
+
+void BMC::inlineFunctionCall(FunctionCall const& _funCall)
+{
+	solAssert(shouldInlineFunctionCall(_funCall, currentScopeContract(), m_currentContract), "");
+
+	auto funDef = functionCallToDefinition(_funCall, currentScopeContract(), m_currentContract);
+	Expression const* calledExpr = &_funCall.expression();
+	auto funType = dynamic_cast<FunctionType const*>(calledExpr->annotation().type);
+	std::vector<ASTPointer<Expression const>> arguments = _funCall.sortedArguments();
+
+	inlineFunctionCall(funDef, _funCall, calledExpr, funType, arguments);
```

Comment:
Here we extract only the name of the function to `calledExpr` from the whole function call expression, see my comment above about difference in what `calledExpr` represents.

Reply:
yes I agree, the callstack object needs to be the one of type `FunctionCall*` in case of an actual function call

Reply:
@leonardoalt, to make sure, the current code is ok in that regard?

Reply:
@leonardoalt , I am still confused about "callstack object".  This should be the expression that represents the function call in the caller?

For me the confusing bit here is the `calledExpr`. In the case of an actual function call, this will be the expression representing the name of the function (because it is `FunctionCall::expression`).
Why would we refer to the expressions of the function name as `calledExpr`? It is not the expression that represents the function call, only the name of the function.